n3fit/src/n3fit/layers/DIS.py

```diff
@@ -4,6 +4,16 @@
     This layer produces a DIS observable, which can consists of one or more fktables.
     The rationale behind this layer is to keep all required operation in one single place
     such that is easier to optimize or modify.
+
+    Comment on the branching based on the number of replicas:
+        This is purely for performance, masking the PDF is more efficient than padding the fk table
+        for one replica, and so is tensordot over einsum.
+
+        Some timings done on snellius using tensorflow 2.15.0 and varying these 2 factors:
+            | CPU\GPU | einsum | tensordot |
+            | -- | -- | -- |
+            | mask pdf | -  | 92 \ 65 |
+            |mask fk | 330 \ 53 \  | 177 \ 53 |
```

Comment:
```suggestion
            |mask fk | 330 \ 53 \  | 177 \ 53 |
            
            These timings are all for one replica.
            
            Crucially, `einsum` is a requirement of the multireplica case, while `tensordot` gives a benefit of a factor of 2x for the single replica case.
            Since this branching is required anyhow,
             by masking the PDF for 1 replica instead of padding the fktable we get an extra factor of x2
```

Adding some extra info otherwise in two months I'll see this code and I'll think "why... but the gpu with tensordot is perfect, why do we care about einsum" x)